ethree-common/src/main/java/com/virgilsecurity/android/common/manager/GroupManager.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2015-2019, Virgil Security, Inc.
+ *
+ * Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     (1) Redistributions of source code must retain the above copyright notice, this
+ *     list of conditions and the following disclaimer.
+ *
+ *     (2) Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ *     (3) Neither the name of virgil nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.virgilsecurity.android.common.manager
+
+import com.virgilsecurity.android.common.exception.GroupException
+import com.virgilsecurity.android.common.exception.GroupNotFoundException
+import com.virgilsecurity.android.common.model.Group
+import com.virgilsecurity.android.common.model.GroupInfo
+import com.virgilsecurity.android.common.model.RawGroup
+import com.virgilsecurity.android.common.model.Ticket
+import com.virgilsecurity.android.common.storage.cloud.CloudTicketStorage
+import com.virgilsecurity.android.common.storage.local.GroupStorageFile
+import com.virgilsecurity.android.common.storage.local.KeyStorageLocal
+import com.virgilsecurity.common.model.Data
+import com.virgilsecurity.sdk.cards.Card
+import com.virgilsecurity.sdk.crypto.VirgilCrypto
+
+/**
+ * GroupManager
+ */
+class GroupManager(
```

Comment:
Shouldn't here and at all internal classes be proper access modifier in order to not expose those classes to user?

Reply:
Leaving as NOT internal till next version, because of language specific stuff, won't affect security or backwards compatability

Reply:
How it's planned to be fixed? Moving it to internal later? How could that not break compatibility?

Reply:
First thing coming up to mind - we'll create a gradle task that will copy all sources to test module, so `internal` modifier will work for tests after that as they will be in the same module with sources for test phase.
It won't break compatibility because visibility modifier will be changed only for non-public classes, that are not supposed to be used while using e3kit.

Reply:
I guess, one of main purposes of access modifiers is to dictate which classes/methods is supposed to be used and which is not. We can't be sure nobody will start using it e.g. for testing purposes.

Reply:
Yes, I agree with that. 
Actually the only problem - I cannot give estimates for that approach with copying sources to tests module. Maybe it'll take a few hours, maybe a few days, because of lack of information for this topic.